torchtune/modules/transformer.py

```diff
@@ -158,12 +160,22 @@ def setup_caches(self, max_batch_size: int, dtype: torch.dtype) -> None:
             torch.ones(self.max_seq_len, self.max_seq_len, dtype=torch.bool)
         )
 
-    def forward(self, tokens: Tensor, input_pos: Optional[Tensor] = None) -> Tensor:
+    def forward(
+        self,
+        tokens: Tensor,
```

Comment:
I would be in favor of having the optional params be keyword-only to avoid confusion in the future.

cc: @ebsmothers